unit_tests/RecordUnitTests.d

```diff
@@ -524,3 +524,25 @@ D");
 
 }
 
+@("issue #foo - translate record with name from macro")
+unittest {
+    assertTranslates(
+    q{
+
+#define __FSID_T_TYPE struct { int __val[2]; }
+typedef  __FSID_T_TYPE __fsid_t;
+typedef __fsid_t fsid_t;
+
+},
+    q{extern (C):
```

Comment:
And in Emacs it's not just highlighting but formatting.

It's great that TextMate users will benefit from this, but I'm not one of them and am unlikely to ever be. I _could_ try to do the same for Emacs, but `d-mode` is an offshooot of the hairy mess that is `c-mode` and well, I'd really rather stay away from it.

I'll change it, but I'm not going to like it. `q{}` is part of the D language, `q"D` isn't.